lib/new_relic/agent/configuration/dotted_hash.rb

```diff
@@ -36,6 +36,7 @@ def self.symbolize(hash)
         def dot_flattened(nested_hash, names = [], result = {})
           nested_hash.each do |key, val|
             next if val.nil?
+
```

Comment:
`next` is interesting to me as a guard clause! I guess here we're exiting early from a particular iteration